<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1066</width>
    <height>653</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(20, 20, 20);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">/* Modern Application Style */
QMainWindow, QDialog {
    background-color: #121212;
    color: #f5f5f5;
    font-family: &quot;Inter&quot;, &quot;Roboto&quot;, &quot;Segoe UI&quot;, sans-serif;
    font-size: 10pt;
}

/* PyQtGraph Styling */
QWidget[class=&quot;PlotWidget&quot;] {
    background-color: #1e1e1e;
    border: 1px solid #333333;
    border-radius: 8px;
}

/* For plot background within the PyQtGraph widget */
/* Note: Apply these in Python code:
   plot.setBackground('#1e1e1e')
   plot.showGrid(x=True, y=True, alpha=0.2)
   plot.getAxis('bottom').setPen(color='#808080')
   plot.getAxis('left').setPen(color='#808080')
*/

/* Push Buttons */
QPushButton {
    background-color: #2563eb;
    color: white;
    border: none;
    padding: 8px 16px;
    border-radius: 8px;
    font-weight: 600;
    min-width: 90px;
    text-transform: uppercase;
    font-size: 9pt;
    letter-spacing: 0.5px;
}

QPushButton:hover {
    background-color: #3b82f6;
    transition: background-color 0.3s;
}

QPushButton:pressed {
    background-color: #1d4ed8;
}

QPushButton:disabled {
    background-color: #333333;
    color: #666666;
}

/* Combo Boxes */
QComboBox {
    border: 2px solid #333333;
    border-radius: 8px;
    padding: 8px 12px;
    background-color: #1e1e1e;
    color: #ffffff;
    selection-background-color: white;
    selection-color: white;
    min-width: 120px;
    font-family: &quot;Inter&quot;, &quot;Roboto&quot;, &quot;Segoe UI&quot;, sans-serif;
    font-size: 10pt;
    font-weight: 500;
}

QComboBox:hover {
    border: 2px solid #2563eb;
}

QComboBox:on {
    border: 2px solid #2563eb;
    background-color:  #1e1e1e;
	color: rgb(255, 255, 255);
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: right;
    width: 24px;
    border-left: none;
}

QComboBox::down-arrow {
    width: 14px;
    height: 14px;
}

QComboBox QAbstractItemView {
    border: 2px solid #333333;
    selection-background-color: #2563eb;
    selection-color: white;
    background-color: #1e1e1e;
    outline: 0px;
    border-radius: 4px;
    padding: 4px;
}

/* Spin Box */
QSpinBox, QDoubleSpinBox {
    border: 2px solid #333333;
    border-radius: 8px;
    padding: 8px 12px;
    background-color: #1e1e1e;
    color: #f5f5f5;
    selection-background-color: #2563eb;
    selection-color: white;
    min-width: 90px;
 font-family: &quot;Inter&quot;, &quot;Roboto&quot;, &quot;Segoe UI&quot;, sans-serif;
    font-size: 10pt;
    font-weight: 500;
}

QSpinBox:hover, QDoubleSpinBox:hover {
    border: 2px solid #2563eb;
}

QSpinBox::up-button, QDoubleSpinBox::up-button {
    subcontrol-origin: border;
    subcontrol-position: top right;
    width: 22px;
    border-left: none;
    border-top-right-radius: 6px;
    background-color: #333333;
    margin-right: 2px;
    margin-top: 2px;
}

QSpinBox::down-button, QDoubleSpinBox::down-button {
    subcontrol-origin: border;
    subcontrol-position: bottom right;
    width: 22px;
    border-left: none;
    border-bottom-right-radius: 6px;
    background-color: #333333;
    margin-right: 2px;
    margin-bottom: 2px;
}

QSpinBox::up-button:hover, QSpinBox::down-button:hover,
QDoubleSpinBox::up-button:hover, QDoubleSpinBox::down-button:hover {
    background-color: #2563eb;
}

QSpinBox::up-arrow, QDoubleSpinBox::up-arrow {
    image: url(up-arrow.png);  /* Replace with path to your arrow icon */
    width: 12px;
    height: 12px;
}

QSpinBox::down-arrow, QDoubleSpinBox::down-arrow {
    image: url(down-arrow.png);  /* Replace with path to your arrow icon */
    width: 12px;
    height: 12px;
}

/* Additional Modern Styling */
QLabel {
    color: #f5f5f5;
 	font-family: &quot;Inter&quot;, &quot;Roboto&quot;, &quot;Segoe UI&quot;, sans-serif;
    font-size: 10pt;
    font-weight: 500;
}

QGroupBox {
    border: 2px solid #333333;
    border-radius: 8px;
    margin-top: 24px;
    font-weight: 600;
    padding-top: 16px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top left;
    padding: 0 10px;
    color: #3b82f6;
    font-size: 11pt;
}

/* Scrollbars */
QScrollBar:horizontal {
    border: none;
    background-color: #1e1e1e;
    height: 8px;
    margin: 0px 0px 0px 0px;
    border-radius: 4px;
}

QScrollBar:vertical {
    border: none;
    background-color: #1e1e1e;
    width: 8px;
    margin: 0px 0px 0px 0px;
    border-radius: 4px;
}

QScrollBar::handle:horizontal, QScrollBar::handle:vertical {
    background-color: #555555;
    border-radius: 4px;
    min-width: 30px;
    min-height: 30px;
}

QScrollBar::handle:horizontal:hover, QScrollBar::handle:vertical:hover {
    background-color: #2563eb;
}

QScrollBar::add-line:horizontal, QScrollBar::sub-line:horizontal,
QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
    border: none;
    background: none;
    width: 0px;
    height: 0px;
}

/* Tooltip */
QToolTip {
    border: 1px solid #333333;
    background-color: #1e1e1e;
    color: #f5f5f5;
    padding: 5px;
    border-radius: 4px;
    opacity: 220;
}

/* Status Bar */
QStatusBar {
    background-color: #121212;
    color: #808080;
    border-top: 1px solid #333333;
}

/* Tab Widget */
QTabWidget::pane {
    border: 2px solid #333333;
    border-radius: 8px;
    top: -2px;
}

QTabBar::tab {
    background-color: #1e1e1e;
    color: #808080;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    padding: 8px 16px;
    margin-right: 4px;
}

QTabBar::tab:selected {
    background-color: #2563eb;
    color: white;
}

QTabBar::tab:hover:!selected {
    background-color: #262626;
    color: #f5f5f5;
}
/* Modern Checkbox Styling for Dark UI */
QCheckBox {
    spacing: 8px;
    color: #f5f5f5;
    font-family: &quot;Inter&quot;, &quot;Roboto&quot;, &quot;Segoe UI&quot;, sans-serif;
    font-size: 10pt;
    font-weight: 500;
    padding: 4px;
}

QCheckBox:hover {
    color: #ffffff;
}

QCheckBox::indicator {
    width: 20px;
    height: 20px;
    border-radius: 4px;
    border: 2px solid #555555;
    background-color: #1e1e1e;
}

QCheckBox::indicator:hover {
    border: 2px solid #3b82f6;
}

QCheckBox::indicator:unchecked {
    background-color: #1e1e1e;
}

QCheckBox::indicator:unchecked:hover {
    background-color: #262626;
    border: 2px solid #3b82f6;
}

QCheckBox::indicator:checked {
    background-color: #2563eb;
    border: 2px solid #2563eb;
    image: url(check.png); /* Replace with path to a white checkmark icon */
}

QCheckBox::indicator:checked:hover {
    background-color: #3b82f6;
    border: 2px solid #3b82f6;
}

QCheckBox::indicator:indeterminate {
    background-color: #4b5563;
    border: 2px solid #4b5563;
    image: url(dash.png); /* Replace with path to a white dash/minus icon */
}

QCheckBox::indicator:indeterminate:hover {
    background-color: #6b7280;
    border: 2px solid #6b7280;
}

QCheckBox:disabled {
    color: #6b7280;
}

QCheckBox::indicator:disabled {
    border: 2px solid #333333;
    background-color: #262626;
}</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="3">
         <widget class="Line" name="line_14">
          <property name="styleSheet">
           <string notr="true">    background-color: #2563eb;
</string>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="PlotWidget" name="plot_second" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="PlotWidget" name="plot_output" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="Line" name="line_13">
          <property name="styleSheet">
           <string notr="true">    background-color: #2563eb;
</string>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="Line" name="line_16">
          <property name="styleSheet">
           <string notr="true">    background-color: #2563eb;
</string>
          </property>
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="PlotWidget" name="plot_original" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="Line" name="line_15">
          <property name="styleSheet">
           <string notr="true">    background-color: #2563eb;
</string>
          </property>
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="PlotWidget" name="plot_hyprid" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>5</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QPushButton" name="pushButton_browse">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_toggle">
          <property name="text">
           <string>Toggle High/Low (Hyprid Image)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_noise">
          <item>
           <property name="text">
            <string>No Noise</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Gaussian Noise</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Salt &amp; Pepper Noise</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Uniform Noise</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_lowpass">
          <item>
           <property name="text">
            <string>LP Filter Type</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Gaussian</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Average</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Median</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_edge">
          <item>
           <property name="text">
            <string>No Edge Detection</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Sobel Mask</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Roberts Mask</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Canny Mask</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Prewitt Mask</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_freq">
          <item>
           <property name="text">
            <string>Freq Domain Filter</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Low Pass Filter</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>High Pass Filter</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>                   Cutoff Frequency</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string> Low</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_lowcutoff">
            <property name="minimumSize">
             <size>
              <width>118</width>
              <height>20</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string> High</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_highcutoff">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>118</width>
              <height>20</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="Line" name="line_18">
            <property name="styleSheet">
             <string notr="true">    background-color: #2563eb;
</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QCheckBox" name="checkBox_global">
              <property name="text">
               <string>Otso Thresh.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox_local">
              <property name="text">
               <string>Local Thresh.</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>  Local Th. Method.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_localmethod">
              <item>
               <property name="text">
                <string>Mean</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Gaussian</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line_17">
            <property name="styleSheet">
             <string notr="true">    background-color: #2563eb;
</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QCheckBox" name="checkBox_equalize">
              <property name="text">
               <string>Equalize             </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox_normalize">
              <property name="text">
               <string>Normalize</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_disthist">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Distribution and Histogram</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_grayscale">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>gray scale</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_reset">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1066</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
